src/structure/structure.ts

```diff
@@ -267,4 +357,258 @@ export class StructureViewer {
+            } else {
+            indexes = this._indexer.from_structure_atom(this._currentList[activeIdx].structure,
+                                                            this._currentList[activeIdx].atom);
+            }
+          this.onselect(indexes, this._active);
+        }
+      }
+    }
+
+    private _setDelay() {
+      if (this._active !== '' ) {
+        return document.getElementById(`chsp-${this._active}-playback-delay`) as HTMLInputElement;
+      } else {
+        throw Error('Active widget is not set.');
+      }
+    }
+
+    /*
+     * Function to resize the grid to `n` widgets
+     */
+    private _resizeGrid(n: number) {
+      if (n < 1) {
+        sendWarning('Cannot have a grid with zero widgets.');
+      } else {
+        if (n > maxWidgets ) {
+        sendWarning(`Widget grid cannot contain more than ${maxWidgets} widgets.`);
+        } else {
+        this._nwidgets = n;
+        this._root = this._setupGrid();
+        }
+      }
+    }
+
+    /*
+     * Function to resize the grid to this._nwidgets + inc widgets
+     */
+    private _resizeGridInc(inc: number) {
+      if (this._nwidgets + inc < 1) {
+        sendWarning('Cannot delete last widget.');
+      } else {
+        if (this._nwidgets + inc > maxWidgets ) {
+        sendWarning(`Widget grid cannot contain more than ${maxWidgets} widgets.`);
+        } else {
+        this._nwidgets = this._nwidgets + inc;
+        this._root = this._setupGrid();
+        }
+      }
+    }
+
+    private _setupCell(cellNo: number, colNum: number, rowNum: number) {
+      if ( cellNo >= this._guids.length) {
+        this._guids.push(generateGUID());
+      }
+
+      const guid = this._guids[cellNo];
+      const cellId = `gi-${guid}`;
+      let cell = document.getElementById(cellId);
+
+      if (cell === null) {
+        cell = document.createElement('div');
+        cell.id = cellId;
+        cell.classList.add('chsp-widget-cell', 'grid-item');
+        cell.style.gridColumn = `${colNum}`;
+        cell.style.gridRow = `${rowNum}`;
+
+        // this was removed because it conflicted with the buttons
+        // cell.onclick = () => {this._setActive(guid)}
+
+        // use HTML5 template to generate a DOM object from an HTML string
+        const activeFlag = document.createElement('div');
+        activeFlag.classList.add('chsp-inactive', 'chsp-circle-button');
+        activeFlag.id = `chsp-activate-${guid}`;
+        this._colors.push(getFarColor(this._colors));
+        activeFlag.style.backgroundColor = this._colors[this._colors.length - 1];
+        activeFlag.onclick = () => { this._setActive(guid); };
+        activeFlag.innerHTML = `<div><span class="tooltiptext">Choose as active</span></div>`;
+        cell.appendChild(activeFlag);
+
+        // use HTML5 template to generate a DOM object from an HTML string
+        const close = document.createElement('div');
```

Comment:
I would use a `button` element instead of a `div`, since it already have most of the right semantics (clickable everywhere, changes pointer styling, ...)